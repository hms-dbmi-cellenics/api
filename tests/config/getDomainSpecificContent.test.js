const getDomainSpecificContent = require('../../src/config/getDomainSpecificContent');
const { ACCOUNT_ID } = require('../../src/api.v2/constants');

describe('getDomainSpecificContent tests', () => {
  // Make a snapshot for each account
  Object.keys(ACCOUNT_ID).forEach((accountId) => {
    it(`check content for ${accountId}`, () => {
      // Set the environment variables
      process.env.AWS_ACCOUNT_ID = ACCOUNT_ID[accountId];
      process.env.NODE_ENV = 'production'; // Since the function checks if env is not 'development'
      process.env.K8S_ENV = 'production';

      const content = getDomainSpecificContent();
      expect(content).toMatchSnapshot(`getDomainSpecificContent for ${accountId}`);
    });
  });

  // Test for the default biomage_private account
  it('check content for the biomage account', () => {
    process.env.AWS_ACCOUNT_ID = 'some_private_account_id';
    process.env.NODE_ENV = 'production'; // Since the function checks if env is not 'test'

    const defaultContent = getDomainSpecificContent();
    expect(defaultContent).toMatchSnapshot('getDomainSpecificContent for biomage private account');
  });

  // Test for the 'test' account
  it('check content for the test account', () => {
    process.env.AWS_ACCOUNT_ID = 'non_existent_account_id';
    process.env.NODE_ENV = 'test'; // Since the function checks if env is not 'test'

    const defaultContent = getDomainSpecificContent();
    expect(defaultContent).toMatchSnapshot('getDomainSpecificContent for test');
  });
});
