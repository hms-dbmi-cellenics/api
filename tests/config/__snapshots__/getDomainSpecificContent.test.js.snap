// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDomainSpecificContent tests check content for BIOMAGE: getDomainSpecificContent for BIOMAGE 1`] = `
Object {
  "adminEmail": "admin@biomage.net",
  "githubOrganisationName": "biomage-org",
  "middlewareChecks": [Function],
  "moreEmailInfo": "More information about Cellenics can be found at <a href=\\"https://biomage.net\\">biomage.net</a>.<br/><br/>
              If you need help or have any questions, please contact us at hello@biomage.net. <br/><br/>",
  "notificationEmail": "notification@localhost:5000",
}
`;

exports[`getDomainSpecificContent tests check content for HMS: getDomainSpecificContent for HMS 1`] = `
Object {
  "adminEmail": "cellenics_admin@listserv.med.harvard.edu",
  "githubOrganisationName": "hms-dbmi-cellenics",
  "middlewareChecks": [Function],
  "moreEmailInfo": "",
  "notificationEmail": "alex_pickering@hms.harvard.edu",
}
`;

exports[`getDomainSpecificContent tests check content for the biomage account: getDomainSpecificContent for biomage private account 1`] = `
Object {
  "adminEmail": "admin@biomage.net",
  "githubOrganisationName": "biomage-org",
  "middlewareChecks": [Function],
  "moreEmailInfo": "More information about Cellenics can be found at <a href=\\"https://biomage.net\\">biomage.net</a>.<br/><br/>
              If you need help or have any questions, please contact us at hello@biomage.net. <br/><br/>",
  "notificationEmail": "notification@localhost:5000",
}
`;

exports[`getDomainSpecificContent tests check content for the test account: getDomainSpecificContent for test 1`] = `
Object {
  "adminEmail": "admin@example.com",
  "githubOrganisationName": "org",
  "middlewareChecks": [Function],
  "moreEmailInfo": "<div>test hi info</div>",
  "notificationEmail": "some@email.com",
}
`;
